### PR TITLE
WindowProxy: also attempt defining accessor indexed properties

### LINKED ISSUE #5599: Also test accessor properties in WindowProxy [[DefineOwnProperty]]
@annevk, it appears per https://bugs.chromium.org/p/chromium/issues/detail?id=695385#c13 at least that Chrome has bugs with accessor properties that don't manifest for data properties. So adding a few lines to https://github.com/w3c/web-platform-tests/commit/8bb5fdc92acf9479599db3827e9cd71dc8f5aa92 that also try accessor properties would be ideal.

### PR DESCRIPTION
**Pull request by annevk**
Fixes #5599.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/5601)

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5601/)

File: html/browsers/the-window-object/window-indexed-properties-strict.html [hunk 1]
```diff
@@ -24,6 +24,8 @@
     window[0] = "foo";
   });
   assert_throws(new TypeError(), () => Object.defineProperty(window, 0, { value: "bar" }))
+  assert_throws(new TypeError(), () => Object.defineProperty(window, 0, { get() { return "baz" } }))
+  assert_throws(new TypeError(), () => Object.defineProperty(window, 1, { set() { return "qux" } }))
```
Comment (domenic) [inline html/browsers/the-window-object/window-indexed-properties-strict.html]: Has to be `set(v)` or else it'll be a syntax error.

File: html/browsers/the-window-object/window-indexed-properties-strict.html [hunk 2]
```diff
@@ -24,6 +24,8 @@
...
   assert_equals(window[0],
                 document.getElementsByTagName("iframe")[0].contentWindow);
```
Comment (domenic) [inline html/browsers/the-window-object/window-indexed-properties-strict.html]: Should we be asserting window[1] is undefined, like the next test does? I'm kind of confused about the test split honestly.

**Review by domenic — CHANGES_REQUESTED** [review]

File: html/browsers/the-window-object/window-indexed-properties-strict.html (hunk 1 above)
Comment (annevk) [inline html/browsers/the-window-object/window-indexed-properties-strict.html]: Hmm, so how did it pass locally?

File: html/browsers/the-window-object/window-indexed-properties.html [hunk 3]
```diff
@@ -20,12 +20,17 @@
 test(function() {
   window[0] = "foo";
   assert_throws(new TypeError(), () => Object.defineProperty(window, 0, { value: "bar" }))
+  assert_throws(new TypeError(), () => Object.defineProperty(window, 0, { get() { return "baz" } }))
+  assert_throws(new TypeError(), () => Object.defineProperty(window, 0, { set() { return "quz" } }))
```
Comment (domenic) [inline html/browsers/the-window-object/window-indexed-properties.html]: Missed one

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/223230812)

# Lint #

Passed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/223230815)

# Firefox (nightly channel) #
Testing web-platform-tests at revision a5c7b61af25144e51dc686e786b91b6440d747da
Using browser at version BuildID 20170417100236; SourceStamp c697e756f738ce37abc56f31bfbc48f55625d617
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5601/html/browsers/the-window-object/window-indexed-properties-strict.html">/html/browsers/the-window-object/window-indexed-properties-strict.html</a></summary>


|                          Subtest                          | Results | Messages |
|-----------------------------------------------------------|---------|----------|
|                                                           | OK      |          |
| `Indexed properties of the window object (strict mode)`   | PASS    |          |
| `Indexed properties of the window object (strict mode) 1` | PASS    |          |
| `Indexed properties of the window object (strict mode) 2` | PASS    |          |
| `Indexed properties of the window object (strict mode) 3` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5601/html/browsers/the-window-object/window-indexed-properties.html">/html/browsers/the-window-object/window-indexed-properties.html</a></summary>


|                            Subtest                            | Results | Messages |
|---------------------------------------------------------------|---------|----------|
|                                                               | OK      |          |
| `Indexed properties of the window object (non-strict mode)`   | PASS    |          |
| `Indexed properties of the window object (non-strict mode) 1` | PASS    |          |
| `Indexed properties of the window object (non-strict mode) 2` | PASS    |          |
| `Indexed properties of the window object (non-strict mode) 3` | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/223230818)

# Chrome (unstable channel) #
Testing web-platform-tests at revision a5c7b61af25144e51dc686e786b91b6440d747da
Using browser at version 59.0.3067.0 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5601/html/browsers/the-window-object/window-indexed-properties-strict.html">/html/browsers/the-window-object/window-indexed-properties-strict.html</a></summary>


|                          Subtest                          | Results |                                       Messages                                       |
|-----------------------------------------------------------|---------|--------------------------------------------------------------------------------------|
|                                                           | OK      |                                                                                      |
| `Indexed properties of the window object (strict mode)`   | PASS    |                                                                                      |
| `Indexed properties of the window object (strict mode) 1` | FAIL    | `assert_throws: function "function () {\n    window[0] = "foo";\n  }" did not throw` |
| `Indexed properties of the window object (strict mode) 2` | FAIL    | `assert_throws: function "function () {\n    window[1] = "foo";\n  }" did not throw` |
| `Indexed properties of the window object (strict mode) 3` | PASS    |                                                                                      |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5601/html/browsers/the-window-object/window-indexed-properties.html">/html/browsers/the-window-object/window-indexed-properties.html</a></summary>


|                            Subtest                            | Results |                                              Messages                                              |
|---------------------------------------------------------------|---------|----------------------------------------------------------------------------------------------------|
|                                                               | OK      |                                                                                                    |
| `Indexed properties of the window object (non-strict mode)`   | PASS    |                                                                                                    |
| `Indexed properties of the window object (non-strict mode) 1` | FAIL    | `assert_throws: function "() => Object.defineProperty(window, 0, { value: "bar" })" did not throw` |
| `Indexed properties of the window object (non-strict mode) 2` | FAIL    | `assert_throws: function "() => Object.defineProperty(window, 1, { value: "bar" })" did not throw` |
| `Indexed properties of the window object (non-strict mode) 3` | PASS    |                                                                                                    |

</details>
</details>

File: html/browsers/the-window-object/window-indexed-properties.html (hunk 3 above)
Comment (annevk) [inline html/browsers/the-window-object/window-indexed-properties.html]: Still though, it's not a SyntaxError in Firefox, is that a bug?

Comment (domenic) [inline html/browsers/the-window-object/window-indexed-properties.html]: Oh, I see my issue. It is the object literal syntax `{ set x() { return "baz"; } }` that is a SyntaxError. In defineProperty, it's just a normal object with a key named "set", which has no syntactic requirements.

Comment (annevk) [inline html/browsers/the-window-object/window-indexed-properties.html]: So do you still want it fixed?

Comment (domenic) [inline html/browsers/the-window-object/window-indexed-properties.html]: Well, it'd be nice to be consistent, but it no longer matters.

My above comment about asserting window[1] still stands though.

Comment (annevk) [inline html/browsers/the-window-object/window-indexed-properties.html]: That already happens in the second test, no?

Comment (domenic) [inline html/browsers/the-window-object/window-indexed-properties.html]: Hmm, maybe it was a copypasta you fixed. OK, LGTM.

**Pull request merged**